mapadroid/patcher/__init__.py

```diff
@@ -37,9 +36,23 @@ def __init__(self, args, data_manager):
         self.dbwrapper = self.data_manager.dbc
         self._schema_updater: DbSchemaUpdater = self.dbwrapper.schema_updater
         self._madver = list(MAD_UPDATES.keys())[-1]
-        self.__get_installed_version()
-        if self.__update_required():
-            self.__update_mad()
+        self._installed_ver = None
+        self.__validate_versions_schema()
+        # If the versions table does not exist we have to install the schema.  When the schema is set, we are then on
+        # the latest table so none of these checks are required
+        if self._installed_ver is None:
+            self.__convert_mappings()
+            self.__get_installed_version()
+            # TODO - When should these be executed?  Can the be added somewhere and ignored?
```

Comment:
I'm not comfortable adding in a TODO in a PR - the position they are in looks ok as long as none of the patches ever expect these tables and columns to be in a different state than "newest" - if they do, they shouldn't be called "unversioned" (obviously)

Reply:
That to-do needed to be dropped. During initial development I dropped that in to remind myself to validate where to put them. I'll remove that to-do line